light-client/relayer/tests/block_processing.rs

```diff
@@ -0,0 +1,225 @@
+        let (node_id, signer) = test_node_id_and_signer(idx);
+        signers.push(signer);
+        quorum_nodes.push(QuorumSetMember::Node(HexKeyNodeID {
+            responder_id: node_id.responder_id,
+            public_key: node_id.public_key,
+        }));
+    }
+
+    let config = Config {
+        start_block_index: 1,
+        ledger_db: ledger_db_path,
+        watcher_db: watcher_db_path,
+        admin_listen_uri: None,
+        verifier_config: LightClientVerifierConfig {
+            trusted_validator_set: TrustedValidatorSetConfig {
+                quorum_set: QuorumSet {
+                    threshold: 3,
+                    members: quorum_nodes.clone(),
+                },
+            },
+            trusted_validator_set_start_block: 1,
+            historical_validator_sets: Default::default(),
+            known_valid_block_ids: BTreeSet::default(),
+        },
+    };
+
+    log::info!(logger, "Starting relayer");
+    let sender = TestSender {
+        logger: logger.clone(),
+        sent: Default::default(),
+    };
+
+    let verifier = LightClientVerifier::from(config.verifier_config.clone());
+
+    let mut relayer = Relayer::new(
+        config,
+        ledger.clone(),
+        watcher.clone(),
+        sender.clone(),
+        verifier,
+        logger.clone(),
+    );
+
+    // Do happy path testing where there is a burn txo
+    let burn_txo = TxOut::new_with_memo(
+        BLOCK_VERSION,
+        Amount::new(100, 2.into()),
+        &burn_address(),
+        &FromRandom::from_random(&mut rng),
+        EncryptedFogHint::fake_onetime_hint(&mut rng),
+        |_ctxt| Ok(BurnRedemptionMemo::new([7u8; 64]).into()),
+    )
+    .unwrap();
+
+    let random_txo = TxOut::new(
+        BLOCK_VERSION,
+        Amount::new(100, 2.into()),
+        &PublicAddress::from_random(&mut rng),
+        &FromRandom::from_random(&mut rng),
+        EncryptedFogHint::fake_onetime_hint(&mut rng),
+    )
+    .unwrap();
+
+    log::info!(logger, "Adding block");
+
+    let (block, block_contents, block_metadata) =
+        append_tx_outs_as_block(&mut ledger, &[burn_txo.clone(), random_txo], signers);
+
+    log::info!(logger, "Adding block to watcher");
+
+    // Add this stuff to the watcher
+    for (idx, block_metadata) in block_metadata.iter().enumerate() {
+        let block_data = BlockData::new(
+            block.clone(),
+            block_contents.clone(),
+            None,
+            Some(block_metadata.clone()),
+        );
+        let url = Url::from_str(&format!("https://node{idx}.test.com:8433")).unwrap();
+
+        watcher.add_block_data(&url, &block_data).unwrap();
+    }
+
+    let mut retries = 100;
+    loop {
+        if retries == 0 {
+            panic!("relayer message not received");
+        } else {
+            let records = sender.sent.lock().unwrap();
+            if records.len() > 0 {
+                assert_eq!(records.len(), 1);
+                let burn_record = records[0].clone();
```

Comment:
Nit: I think you can just borrow
```suggestion
                let burn_record = &records[0];
```